### PR TITLE
DRY (don't repeat yourself)

### LINKED ISSUE #22231: build: obsolete cmake warning
Sorry to post warning issue

##### System information (version)

- OpenCV => recent 3.4, master, next
- Operating System / Platform => Ubuntu 18.04 (Aarch64) Jetson AGX Xavier
- Compiler => GCC 7.5.0
- CMake => 3.23.2

##### Detailed description

When building under above condition, following message appears.

```
-- VTK is not found. Please set -DVTK_DIR in CMake to VTK build directory, or to VTK install subdirectory with VTKConfig.cmake file
CMake Deprecation Warning at 3rdparty/carotene/hal/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at 3rdparty/carotene/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Looking for dlerror in dl
```

It's triggered by this line

https://github.com/opencv/opencv/blob/70492c212789922b8d71f9b06770f4af25af277b/3rdparty/carotene/CMakeLists.txt#L1

I found the related topics on [CMake Kitware forum](https://discourse.cmake.org/t/how-to-fix-cmake-minimum-required-deprecation-warning/2487)

So this is a backward compatibility issue, and CMake is going to drop backward compatibility.
A workaround is to add the the latest supported version of cmake as the message says, but I don't think that's a better solution, since we now need to update this line every time new cmake appears.

Now looking at another aspect of this issue, OpenCV already defines `MIN_VER_CMAKE` in [OpenCVMinDepVersions.cmake](https://github.com/opencv/opencv/blob/70492c212789922b8d71f9b06770f4af25af277b/cmake/OpenCVMinDepVersions.cmake)

https://github.com/opencv/opencv/blob/70492c212789922b8d71f9b06770f4af25af277b/cmake/OpenCVMinDepVersions.cmake#L2

(This line is picked from 3.4 branch)

So no matter what version of CMake is written in carotene/CMakeLists.txt, it doesn't make sense.
In another word, any `cmake_minimum_request` should refer this `MIN_VER_CMAKE` variable to be consistent.

So for this warning message, I propose to reuse `MIN_VER_CMAKE` where ever `cmake_minimum_request` is used.
For the actual cmake version, I think I should not touch.

I'll send a patch later.

##### Steps to reproduce

```
 cmake -DOPENCV_EXTRA_MODULES_PATH=/opencv_contrib/modules \
 -DBUILD_EXAMPLES=true \
 -DOPENCV_ENABLE_NONFREE=true \
 -DWITH_CUDA=ON \
 -DWITH_EIGEN=OFF \
 -DPYTHON2_EXECUTABLE= \
 -DCMAKE_BUILD_TYPE=Release ..
```

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       forum.opencv.org, Stack Overflow, etc and have not found any solution
   
 - [x] I updated to the latest OpenCV version and the issue is still there
   
 - [x] There is reproducer code and related data files: videos, images, onnx, etc

### PR DESCRIPTION
**Pull request by tomoaki0705**
closes #22231 

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/python/CMakeLists.txt [hunk 1]
```diff
@@ -40,7 +40,7 @@ add_subdirectory(python3)
 
 else()  # standalone build
 
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION ${MIN_VER_CMAKE})
```
Comment (alalek) [inline modules/python/CMakeLists.txt:43]: > standalone

`MIN_VER_CMAKE` may be not available for standalone bindings, samples, hal samples.

**Review by alalek — COMMENTED** [review]
Only the first 3 changes are harmless. Others are probably not.

File: modules/python/CMakeLists.txt (hunk 1 above)
Comment (alalek) [reply modules/python/CMakeLists.txt]: Lets bump to 2.8.12 for other cases to avoid warning.

Comment (tomoaki0705) [reply modules/python/CMakeLists.txt]: I took this "other cases" as standalone bindings, samples and hal samples.
I also bumped ones in the "toolchain.cmake" in platform directory, please let me know if that's along with your expectation.

Only remaining `cmake_minimum_required` that I didn't touch is the one in the doc directory

**Review by alalek — APPROVED** [review]
Thank you :+1:

**Pull request merged**